src/python/pants/engine/internals/rule_visitor.py

```diff
@@ -98,53 +91,29 @@ def _get_awaitable(self, call_node: ast.Call) -> AwaitableConstraints:
                 f"Expected either two or three arguments, but got {len(get_args)} arguments."
             )
 
-        output_expr = get_args[0]
-        if not isinstance(output_expr, ast.Name):
-            raise parse_error(
-                "The first argument should be the output type, like `Digest` or `ProcessResult`."
-            )
-        output_type = output_expr
+        output_node = get_args[0]
+        output_type = self._lookup(output_node)
 
-        input_args = get_args[1:]
-        input_types: List[ast.Name]
-        if len(input_args) == 1:
-            input_constructor = input_args[0]
+        input_nodes = get_args[1:]
+        input_types: List[Any]
+        if len(input_nodes) == 1:
+            input_constructor = input_nodes[0]
             if isinstance(input_constructor, ast.Call):
-                if not isinstance(input_constructor.func, ast.Name):
-                    raise parse_error(
-                        f"Because you are using the shorthand form {call_node.func.id}(OutputType, "
-                        "InputType(constructor args)), the second argument should be a top-level "
-                        "constructor function call, like `MergeDigest(...)` or `Process(...)`, rather "
-                        "than a method call."
```

Comment:
These error messages get worse, but :man_shrugging: It'll still list the incorrect value and file/line.

Reply:
What are they now?